packages/input/src/index.ts

```diff
@@ -50,7 +58,11 @@ export default createPrompt<string, InputConfig>((config, done) => {
     if (isEnterKey(key)) {
       const answer = value || defaultValue;
       setStatus('loading');
-
+      if (pattern && !pattern.test(answer)) {
+        setError(patternError || 'Invalid input update');
+        setStatus('idle');
+        return;
+      }
       const isValid =
         required && !answer ? 'You must provide a value' : await validate(answer);
       if (isValid === true) {
```

Comment:
There's a question of which validation should occur first when both are present... This open 2 questions:

* Should we allow both to be set or not?
* If we do allow both, which should run first?